### DISCUSSION TITLE
NDJSON special support

### DISCUSSION THREAD

**Discussion by phretor** [dadoonet/fscrawler#1113] [Ideas]
Hello!

I feel that with FSCrawler my search for the perfect personal search engine is about to come to a successful end: it's exactly what I've been looking for.

I use NDJSON a lot to lump together different JSON files and reduce the open-close file requests at query time, and I was wondering if FSCrawler is able to handle it as a special case of JSON file, for which I see there is support already.

Speaking in ElasticSearch terminology, it would probably be mapped as a bulk request.

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#docs-bulk-api-desc (see here for NDJSON and bulk)

If that doesn't work out of the box, I'm happy to contribute some code to implement it, if someone could pinpoint the best place to plug this functionality in.

Thanks!

**Comment by dadoonet** [comment]
Hey @phretor 

Thanks for the proposal. That sounds a good idea to me.
I believe it should be a similar option as https://fscrawler.readthedocs.io/en/latest/admin/fs/local-fs.html#indexing-json-docs

```yaml
name: "test"
fs:
  ndjson_support: true
```

About the implementation, you can start from here I think.

https://github.com/dadoonet/fscrawler/blob/a6be17811c0094dec5eb80b88790dc1a48083c7d/core/src/main/java/fr/pilato/elasticsearch/crawler/fs/FsParserAbstract.java#L454-L468

The XML support is more helpful to get inspiration from I believe:

https://github.com/dadoonet/fscrawler/blob/a6be17811c0094dec5eb80b88790dc1a48083c7d/core/src/main/java/fr/pilato/elasticsearch/crawler/fs/FsParserAbstract.java#L469-L484

May be copying the "skeleton" from this class would help?

https://github.com/dadoonet/fscrawler/blob/master/tika/src/main/java/fr/pilato/elasticsearch/crawler/fs/tika/XmlDocParser.java

LMK if you need more information. And thanks a lot for proposing your help!
